operations/UAS_user_and_admin_topics/Troubleshoot_Broker_SSSD_Cant_Use_sssd_conf.md

```diff
@@ -0,0 +1,151 @@
+```
+
+### Problem Explanation
+
+In the current release of UAS, the `default` Service Account on the `uas` namespace in Kubernetes is bound to a Cluster Role that uses a Pod Security Policy that defines a specific `fsGroup` range and a `MustRunAs` rule instead of simly using the `RunAsAny` rule.  Because of the way Kubernetes handles volumes, when a volume containing a Secret or a ConfigMap is mounted on a Kubernetes pod with an `fsGroup` rule that is not `RunAsAny` in the Pod Security Policy, the requested mode of the volume is adjusted to something Kubernetes deems more appropriate.  In this case, the requested mode (decimal 384 or octal 600) becomes octal 640 instead in the mounted volume.  Unfortunately, SSSD requires that this mode be octal 600 or it will refuse to use the configuration file.
+
+### Workaround
+
+While this problem will be resolved in an upcoming release of UAS, if this behavior occurs, it is necessary to create a new Pod Security Policy and a Cluster Role using that Pod Security Policy, then change the existing Cluster Role Binding to bind the new Cluster Role instead of the one it currently uses.  The following procedure does that.
+
+First, verify that the system is set up the same way as the system on which this workaround was prepared.  To do that, list the Cluster Role Binding named `uas-default-psp` and determine what Cluster Role it is bound to:
+
+```
+ncn-m001:~ # kubectl get clusterrolebindings uas-default-psp -o yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+
+...
+
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: restricted-transition-net-raw-psp
+subjects:
+- kind: ServiceAccount
+  name: default
+  namespace: uas
+```
+
+Notice the `roleRef` setting binds the `restricted-transition-net-raw-psp` Cluster Role. That Cluster Role uses the `restricted-transition-net-raw-psp` Pod Security Policy, which is used by several CSM services, but does not work for Broker UAIs.
+
+Assuming this configuration is present, it is necessary to create a new Pod Security Policy called `uas-default-psp`, then create a Cluster Role called `uas-default-psp` that uses the new Pod Security Policy and finally a change the Cluster Role Binding called `uas-defaul-psp` to bind the new Cluster Role to the `default` Service Account in the `uas` namespace.  Since Kubernetes does not allow simply updating the existing `uas-default-psp` Cluster Role Binding the existing one must be removed and replaced.  To remove the existing Cluster Role Binding:
+
+```
+ncn-m001:~ # kubectl delete clusterrolebindings uas-default-psp
+```
+
+Now prepare a YAML file containing the new Kubernetes objects:
+
+```
+ncn-m001:~ # cat << EOF > /tmp/uas-default-psp.yaml
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: uas-default-psp
+spec:
+  allowPrivilegeEscalation: true
+  allowedCapabilities:
+  - NET_ADMIN
+  - NET_RAW
+  allowedHostPaths:
+  - pathPrefix: /lustre
```

Comment:
Where did this allowedHostPaths list come from? I'm a little surprised to see variations of "lustre" in this list and want to understand if this is needed or if we are creating a restrictive list.

Reply:
This came from the original PSP that was being used for the `uas` namespace.  I tried to make minimal changes to that to avoid creating security issues.